### PR TITLE
sign-in updates

### PR DESCRIPTION
**Pull request by davidmu1**

### CODE REVIEW

File: api-reference/beta/api/signin-get.md [hunk 1]
```diff
@@ -44,27 +40,30 @@ This method supports OData query parameters to help customize the response. For
 
 | Name      |Description|
 |:----------|:----------|
-| Authorization  | Bearer {code}|
+| Authorization | Bearer {token} |
+| Content-type | application/json |
```
Comment (dkershaw10) [inline api-reference/beta/api/signin-get.md]: Remove the second line.  This is only required when there's a payload (so POST, PATCH and PUT)

File: api-reference/beta/api/signin-get.md [hunk 2]
```diff
@@ -11,21 +11,17 @@ ms.prod: "microsoft-identity-platform"
 
 [!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
 
-Retrieve a specific Azure AD user sign-in event for your tenant. Sign-ins that are interactive in nature (where a username/password is passed as part of authorization token) and successful federated sign-ins are currently included in the sign-in logs.
-
+Get a [signIn](../resources/signin.md) object that contains a specific user sign-in event for your tenant. This includes sign-ins where a user is asked to enter a username or password, and session tokens.
```
Comment (dkershaw10) [inline api-reference/beta/api/signin-get.md]: I actually think the second sentence is worse than before.  What does session tokens mean?

File: api-reference/beta/api/signin-list.md [hunk 3]
```diff
@@ -11,82 +11,102 @@ ms.prod: "microsoft-identity-platform"
 
 [!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
 
-Retrieve the Azure AD user sign-ins for your tenant. Sign-ins that are interactive in nature (where a username/password is passed as part of authorization token) and successful federated sign-ins are currently included in the sign-in logs.  The most recent signIns are returned first.
+Get a list of [signIn](../resources/signin.md) objects. The list contains the user sign-ins for your Azure Active Directory tenant. Sign-ins where a username and password are passed as part of authorization token, and successful federated sign-ins are currently included in the sign-in logs. The most recent sign-ins are returned first.
```
Comment (dkershaw10) [inline api-reference/beta/api/signin-list.md]: This explanation is not consistent with the Get topic.

File: api-reference/beta/api/signin-list.md [hunk 4]
```diff
@@ -11,82 +11,102 @@ ms.prod: "microsoft-identity-platform"
...
-|Application | Directory.Read.All  | 
+| Permission type | Permissions (from least to most privileged) |
+|:--------------- |:------------------------------------------- |
+| Delegated (work or school account) | AuditLog.Read.All, Directory.Read.All |
+| Delegated (personal Microsoft account) | Not supported |
+| Application | AuditLog.Read.All, Directory.Read.All | 
 
-
-In addition, apps must be [properly registered](https://docs.microsoft.com/azure/active-directory/active-directory-reporting-api-prerequisites-azure-portal) to Azure AD.
+In addition, apps must be [properly registered](https://docs.microsoft.com/azure/active-directory/active-directory-reporting-api-prerequisites-azure-portal) to Azure Active Directory.
```
Comment (dkershaw10) [inline api-reference/beta/api/signin-list.md]: Personally I find the linked topic rather confusing.  It mixes consenting to permissions for Microsoft Graph and Azure AD Graph, and delegated and application permissions.  What's in that linked doc does not align with what is required above.

File: api-reference/beta/api/signin-list.md [hunk 5]
```diff
@@ -11,82 +11,102 @@ ms.prod: "microsoft-identity-platform"
...
+| servicePrincipalName | eq, startswith |
+| userAgent | eq, startswith |
+| alternateSignInName | eq, startswith |
 
+## Request headers
+
+| Name      |Description|
+|:----------|:----------|
+| Authorization | Bearer {token} |
+| Content-type | application/json |
```
Comment (dkershaw10) [inline api-reference/beta/api/signin-list.md]: Remove this line.  Not used in a GET operation

File: api-reference/beta/api/signin-list.md [hunk 6]
```diff
@@ -105,124 +125,217 @@ GET https://graph.microsoft.com/beta/auditLogs/signIns
...
+#### Request
+
+The following is an example of the request.
+
+<!-- {
+  "blockType": "request",
+  "name": "get_signins_2"
+}-->
+```msgraph-interactive
+GET https://graph.microsoft.com/beta/auditLogs/signIns/{id}
```
Comment (dkershaw10) [inline api-reference/beta/api/signin-list.md]: I thought this was a **list** topic?

File: api-reference/beta/resources/authenticationdetail.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+author: "davidmu1"
+ms.prod: "microsoft-identity-platform"
+doc_type: "resourcePageType"
+---
+
+# authenticationDetail resource type
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Provides the authentication details for a user sign-in, such as multi-factor authentication (MFA) information and PTA/PHS details.
```
Comment (dkershaw10) [inline api-reference/beta/resources/authenticationdetail.md]: PTA? PHS?  Do we not need the full forms?

File: api-reference/beta/resources/keyvalue.md [hunk 8]
```diff
@@ -1,50 +1,51 @@
 ---
 title: "keyValue resource type"
-description: "Here is a JSON representation of the resource."
+description: "Provides additional relevant information about the sign-in request"
```
Comment (dkershaw10) [inline api-reference/beta/resources/keyvalue.md]: So this is likely to be a shared type/resource that is used across multiple services.
I think you need to make the info in here much more generic - it's a basic key/value pair type that can be used in multiple different ways.  You can proved examples, but I wouldn't make this sign-in audit specific.

File: api-reference/beta/resources/networkLocationDetail.md [hunk 9]
```diff
@@ -1,50 +1,51 @@
...
 ## Properties
-| Property	   | Type	|Description|
-|:---------------|:--------|:----------|
-|networkType|networkType|Provides the type of the network. The possible values are `intranet`, `extranet`, `namedNetwork`, and `trusted`.|
-|networkNames|String collection|Names of the network.|
 
+| Property     | Type        | Description |
+|:-------------|:------------|:------------|
+|networkNames|String collection|Provides the name of the network used when signing in.|
+|networkType|String| Provides the type of network used when signing in. Possible values are: `intranet`, `extranet`, `namedNetwork`, `trusted`, `unknownFutureValue`.|
```
Comment (dkershaw10) [inline api-reference/beta/resources/networkLocationDetail.md]: If this is a string rather than an enum, please remove `unknownFutureValue`.  If it is an enum then there should be a type name (linking to the enum file?) and the enums file should be updated.

File: api-reference/beta/resources/signin.md [hunk 10]
```diff
@@ -24,36 +24,43 @@ Provides details about user or application sign-in activity in your directory.
...
-|userDisplayName|String|Indicates the display Name of the User.|
-|userId|String|Indicates the userId of the user.|
-|userPrincipalName|String|Indicates the UPN of the user.|
-|resourceDisplayName|String|Indicates the name of the resource that the user signed into|
-|resourceId|String|Indicates the Id of the resource that the user signed into.|
-|authenticationMethodsUsed|String|Indicates the list of Authentication methods used|
+|appDisplayName|String|The application name displayed in the Azure Portal.|
+|appId|String|The application identifier in Azure Active Directory.|
+|clientAppUsed|String|The legacy client used for sign-in activity. For example, Browser, Exchange Active Sync,Modern clients, IMAP, MAPI, SMTP, or POP.|
+|appliedConditionalAccessPolicies|[conditionalAccessPolicy](conditionalaccesspolicy.md) collection|A list of conditional access policies that are triggered by the corresponding sign-in activity.|
```
Comment (dkershaw10) [inline api-reference/beta/resources/signin.md]: Where are you getting this information from?  I'm concerned that the rest of the properties here may have errors.
The type here should be **appliedConditionalAccessPolicy**.  conditionalAccessPolicy is the actual policy definition type and is different from this.

File: api-reference/beta/resources/signin.md [hunk 11]
```diff
@@ -24,36 +24,43 @@ Provides details about user or application sign-in activity in your directory.
...
+|authenticationMethodsUsed|String collection|The authentication methods used. Possible values: `SMS`, `Authenticator App`, `App Verification code`, `Password`, `FIDO`, `PTA`, or `PHS`.|
+|authenticationProcessingDetails|[keyValue](keyvalue.md) collection|Additional authentication processing details, such as the agent name in case of PTA/PHS or Server/farm name in case of federated authentication.|
+|networkLocationDetails|[networkLocationDetail](networklocationdetail.md) collection|The network location details, such as IP address, location of the sign-in, the type of network used, and its names. Possible values: `Named Netowrk`, `Extranet`, `Intranet`, or `Trusted Network`.|
+|servicePrincipalId|String|The application identifier used for sign-in. This field is populated when you are signing in using an application.|
+|servicePrincipalName|String|The application name used for sign-in. This field is populated when you are signing in using an application.|
+|userAgent|String|The user agent information related to sign-in.|
+|alternateSignInName|String|The alternate sign-in identity whenever you use phone number to sign-in.|
```
Comment (dkershaw10) [inline api-reference/beta/resources/signin.md]: Why is this at the end?  Should this be in alphabetical order?

File: api-reference/beta/resources/signin.md [hunk 12]
```diff
@@ -24,36 +24,43 @@ Provides details about user or application sign-in activity in your directory.
...
+|riskLevelDuringSignIn|`riskLevel`|The risk level during sign-in. Possible values: `none`, `low`, `medium`, `high`, `hidden`, or `unknownFutureValue`. The value `hidden` means the user or sign-in was not enabled for Azure AD Identity Protection. **Note:** Details for this property are only available for Azure AD Premium P2 customers. All other customers are returned `hidden`.|
+|riskEventTypes|`riskEventType` collection|The list of risk event types associated with the sign-in. Possible values: `unlikelyTravel`, `anonymizedIPAddress`, `maliciousIPAddress`, `unfamiliarFeatures`, `malwareInfectedIPAddress`, `suspiciousIPAddress`, `leakedCredentials`, `investigationsThreatIntelligence`,  `generic`, or `unknownFutureValue`.|
+|riskState|`riskState`|The risk state of a risky user, sign-in, or a risk event. Possible values: `none`, `confirmedSafe`, `remediated`, `dismissed`, `atRisk`, `confirmedCompromised`, or `unknownFutureValue`.|
+|networkLocationDetails|[networkLocationDetail](networklocationdetail.md) collection|Details about the network location.|
```
Comment (dkershaw10) [inline api-reference/beta/resources/signin.md]: From here down nothing is in alphabetical order.  Please fix.

File: api-reference/beta/resources/signinactivity.md [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+author: "davidmu1"
+ms.prod: "microsoft-identity-platform"
+doc_type: "resourcePageType"
+---
+
+# signInActivity resource type
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+PROVIDE DESCRIPTION HERE
```
Comment (dkershaw10) [inline api-reference/beta/resources/signinactivity.md]: As it says :)

File: api-reference/beta/resources/signinactivity.md [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+
+## Properties
+
+| Property     | Type        | Description |
+|:-------------|:------------|:------------|
+|lastSignInDateTime|DateTimeOffset|The last sign-in date for a specific user. You can use this field to calculate the last time a user signed in to the directory by joining it with the users resource in Microsoft Graph. This field can be used to build reports, such as inactive users. The timestamp represents date and time information using ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2014 would look like this: `'2014-01-01T00:00:00Z'`.|
```
Comment (dkershaw10) [inline api-reference/beta/resources/signinactivity.md]: Remove "by joining it with the users resource in Microsoft Graph".

File: api-reference/beta/resources/signinactivity.md [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+|lastSignInRequestId|String|Request Id of the last sign-in performed by this user.|
```
Comment (dkershaw10) [inline api-reference/beta/resources/signinactivity.md]: Id -> ID

File: api-reference/beta/resources/signinactivity.md [hunk 16]
```diff
@@ -0,0 +1,51 @@
+---
+title: "signInActivity resource type"
```
Comment (dkershaw10) [inline api-reference/beta/resources/signinactivity.md]: TBH - I would have split the signInActivity stuff into a separate PR.  There's some issues getting this to work...
You are also missing the update to the user-list.md topic with the extra couple of examples.

File: api-reference/beta/api/user-list.md [hunk 17]
```diff
@@ -174,6 +175,98 @@ Content-length: 108
...
+      "signInActivity": {
+        "lastSignInDateTime": "2017-07-29T02:16:18Z",
+        "lastSignInRequestId": "90d8b3f8-712e-4f7b-aa1e-62e7ae6cbe96"
+      }
+    }
+  ]
+}
+```
+
+### Example 4: List all users that last signed in on or before a particular date
```
Comment (dkershaw10) [inline api-reference/beta/api/user-list.md]: Actually, I'm thinking we might want to reverse this.
A better query/scenario would be to find all the users that have not signed-in in the last 3 months (inactive users).  So that would be something with a `ge` rather than `le` expression

File: api-reference/beta/api/signin-list.md (hunk 4 above)
Comment (davidmu1) [reply api-reference/beta/api/signin-list.md]: Removed the link.

File: api-reference/beta/api/signin-list.md (hunk 5 above)
Comment (davidmu1) [reply api-reference/beta/api/signin-list.md]: Removed.

File: api-reference/beta/api/signin-list.md (hunk 6 above)
Comment (davidmu1) [reply api-reference/beta/api/signin-list.md]: Removed {id}.

File: api-reference/beta/resources/networkLocationDetail.md (hunk 9 above)
Comment (davidmu1) [reply api-reference/beta/resources/networkLocationDetail.md]: Changed the property type to networkType enumeration. @dhanyahk - Can you please confirm that this is correct?

File: api-reference/beta/resources/signin.md (hunk 12 above)
Comment (davidmu1) [reply api-reference/beta/resources/signin.md]: Updated table order.

File: api-reference/beta/resources/signin.md [hunk 18]
```diff
@@ -73,38 +80,43 @@ Here is a JSON representation of the resource.
...
-  "riskEventTypes": ["String"],
-  "resourceDisplayName": "string",
-  "resourceId": "string",
-  "authenticationMethodsUsed": "string",
+  "servicePrincipalId": "String",
+  "servicePrincipalName": "String",
   "status": {"@odata.type": "microsoft.graph.signInStatus"},
-  "processingTimeInMilliseconds": 12356,
-  "networkLocationDetails": [{"@odata.type": "microsoft.graph.networkLocationDetail"}]
+  "tokenIssuerName": "String",
```
[the remark this answers is not in the document]
Comment (davidmu1) [reply api-reference/beta/resources/signin.md]: These properties are listed in the properties table.

File: api-reference/beta/resources/signinactivity.md (hunk 13 above)
Comment (davidmu1) [reply api-reference/beta/resources/signinactivity.md]: Added the description.

File: api-reference/beta/resources/signinactivity.md (hunk 15 above)
Comment (davidmu1) [reply api-reference/beta/resources/signinactivity.md]: Changed.

File: api-reference/beta/api/user-list.md (hunk 17 above)
Comment (davidmu1) [reply api-reference/beta/api/user-list.md]: @dhanyahk - Can you provide an example?

File: api-reference/beta/api/signin-list.md (hunk 3 above)
Comment (dhanyahk) [reply api-reference/beta/api/signin-list.md]: I didn't change any of this. This is how this has been

File: api-reference/beta/resources/authenticationdetail.md (hunk 7 above)
Comment (dhanyahk) [reply api-reference/beta/resources/authenticationdetail.md]: No, It's how it is called in the string as well

File: api-reference/beta/resources/keyvalue.md (hunk 8 above)
Comment (dhanyahk) [reply api-reference/beta/resources/keyvalue.md]: I think today this is sign in specific. It gives additional authentication information. I am not sure how else to call it since I don't know it's use for other scenarios

File: api-reference/beta/resources/networkLocationDetail.md (hunk 9 above)
Comment (dhanyahk) [reply api-reference/beta/resources/networkLocationDetail.md]: Yes, we can

File: api-reference/beta/resources/signin.md (hunk 10 above)
Comment (dhanyahk) [reply api-reference/beta/resources/signin.md]: This is what the csdl gave. What should I change it to?

File: api-reference/beta/api/user-list.md (hunk 17 above)
Comment (dhanyahk) [reply api-reference/beta/api/user-list.md]: I think @dkershaw10  gave the examples. As he suggested, let's remove this from this PR

**Comment by davidmu1** [comment]
#sign-off
